megamek/src/megamek/server/totalwarfare/TWGameManager.java

```diff
@@ -31789,7 +31790,7 @@ public Vector<Integer> artilleryDamageHex(Coords coords,
             if (!flak) {
                 // Report that damage applied to terrain, if there's TF to damage
                 Hex h = game.getBoard().getHex(coords);
-                if ((h != null) && h.hasTerrainFactor()) {
+                if ((h != null) && h.hasTerrainFactor() && h.hasTerrainFactor() && (altitude == h.getLevel())) {
```

Comment:
`h.hasTerrainFactor()` is in this conditional twice.

Reply:
Okay, I know I pulled that out, one sec...

Reply:
Must have undone the previous fix at some point.  Should be fixed now.